// Copyright 2018 - 2019
// The "project-template" Authors. All rights reserved.
// Use of this source code is governed by the MIT License that can be found in the LICENSE file.

syntax = "proto3";
package ProtoExecutive;

// ------------------------------------------------------
// ---------------- Exec Inbound Message ----------------
// ------------------------------------------------------

enum Process_Id
{
   INVALID_PROCESS = 0;

   LOGGER = 1;
   MODEL = 2;
}

enum Message_Type
{
   INVALID_TYPE = 0;

   VERSION = 1;
   WATCHDOG = 2;
}

message Exec_Inbound_Message
{
   Process_Id proc_id = 1;
   Message_Type message_type = 2;

   oneof payload
   {
      Version_Inbound_Message version = 3;
   }
}

message Version_Inbound_Message
{
   string software_version = 1;
};

// ------------------------------------------------------
// ---------------- Exec Outbound Message ---------------
// ------------------------------------------------------

enum Message_Status
{
   INVALID_STATUS = 0;

   ACCEPT = 1;
   REJECT = 2;
}

message Exec_Outbound_Message
{
   Message_Type message_type = 1;

   oneof payload
   {
      Version_Outbound_Message version = 2;
      Watchdog_Outbound_Message watchdog = 3;
   }
}

message Version_Outbound_Message
{
   Message_Status status = 1;
};

message Watchdog_Outbound_Message
{
   Message_Status status = 1;
};

